opm/simulators/flow/LogOutputHelper.cpp

```diff
@@ -99,132 +126,83 @@ cumulative(const std::size_t reportStepNum) const
-        } else {
-            tmp_values[2] = get("GOPT"); // WellCumDataType::OilProd
-            tmp_values[3] = get("GWPT"); // WellCumDataType::WaterProd
-            tmp_values[4] = get("GGPT"); // WellCumDataType::GasProd
-            tmp_values[5] = get("GVPT"); // WellCumDataType::FluidResVolProd
-            tmp_values[6] = get("GOIT"); // WellCumDataType::OilInj
-            tmp_values[7] = get("GWIT"); // WellCumDataType::WaterInj
-            tmp_values[8] = get("GGIT"); // WellCumDataType::GasInj
-            tmp_values[9] = get("GVIT"); // WellCumDataType::FluidResVolInj
+    for (const auto& gname : this->schedule_.groupNames()) {
+        std::vector<Scalar> values(WellCumDataType::numWCValues, 0.0);
+        std::vector<std::string> names(WellCumDataType::numWCNames, "");
+
+        names[Ix::WellName] = gname;
+
+        if (gname == "FIELD") {
+            values[Ix::OilProd]         = st.get("FOPT", 0.0);
+            values[Ix::WaterProd]       = st.get("FWPT", 0.0);
+            values[Ix::GasProd]         = st.get("FGPT", 0.0);
+            values[Ix::FluidResVolProd] = st.get("FVPT", 0.0);
+            values[Ix::OilInj]          = st.get("FOIT", 0.0);
+            values[Ix::WaterInj]        = st.get("FWIT", 0.0);
+            values[Ix::GasInj]          = st.get("FGIT", 0.0);
+            values[Ix::FluidResVolInj]  = st.get("FVIT", 0.0);
+        }
+        else {
+            values[Ix::OilProd]         = st.get_group_var(gname, "GOPT", 0.0);
+            values[Ix::WaterProd]       = st.get_group_var(gname, "GWPT", 0.0);
+            values[Ix::GasProd]         = st.get_group_var(gname, "GGPT", 0.0);
+            values[Ix::FluidResVolProd] = st.get_group_var(gname, "GVPT", 0.0);
+            values[Ix::OilInj]          = st.get_group_var(gname, "GOIT", 0.0);
+            values[Ix::WaterInj]        = st.get_group_var(gname, "GWIT", 0.0);
+            values[Ix::GasInj]          = st.get_group_var(gname, "GGIT", 0.0);
+            values[Ix::FluidResVolInj]  = st.get_group_var(gname, "GVIT", 0.0);
         }
 
-        this->outputCumulativeReportRecord_(tmp_values, tmp_names);
+        this->outputCumulativeReportRecord_(values, names);
     }
 
-    for (const auto& wname : schedule_.wellNames(reportStepNum)) {
-        const auto& well = schedule_.getWell(wname, reportStepNum);
-        tmp_names[0] = wname; // WellCumDataType::WellName
-        auto wName = static_cast<std::string>(wname);
-        auto get = [&st, &wName](const std::string& vector)
-        {
-            const auto key = vector + ':' + wName;
-            return st.has(key) ? st.get(key) : 0.0;
-        };
+    for (const auto& wname : this->schedule_.wellNames(reportStepNum)) {
+        const auto& well = this->schedule_.getWell(wname, reportStepNum);
+
+        std::vector<Scalar> values(WellCumDataType::numWCValues, 0.0);
+        std::vector<std::string> names(WellCumDataType::numWCNames, "");
```

Comment:
Better use default ctor

Reply:
> Better use default ctor

Agreed.  I've pushed an update to that effect.